applications/HDF5Application/python_scripts/core/operations/model_part.py

```diff
@@ -299,104 +325,132 @@ class PrimalBossakOutput(VariableIO):
+        super().__init__()
+        if isinstance(parameters, ParametersWrapper):
+            parameters = parameters.Get()
+        parameters.AddMissingParameters(self.GetDefaultParameters())
+
+        self.__model_part = model.GetModelPart(parameters["model_part_name"].GetString())
+        self.__io_parameters = parameters["io_settings"]
+
+        # {operation_type, operation_parameters}
+        self.__operations: "list[tuple[type, KratosMultiphysics.Parameters]]" = []
+        for i in range(parameters["list_of_operations"].size()):
+            self.Add(parameters["list_of_operations"][i])
+
+    def Execute(self) -> None:
+        with OpenHDF5File(self.__io_parameters, self.__model_part) as file:
+            for operation, operation_parameters in self.__operations:
+                operation(self.__model_part, operation_parameters, file).Execute()
+
+    def Add(self, operation_parameters: KratosMultiphysics.Parameters) -> None:
+        # Convert input snake case name to the internal camel case name
+        operation_type = self.__SnakeToCamel(operation_parameters["operation_type"].GetString())
+        operation: type = next((op for op in GetSubclasses(IOOperation) if op.__name__ == operation_type), None)
+
+        # Found an operation with a matching name
+        if not (operation is None):
+            operation_parameters.AddMissingParameters(operation.GetDefaultParameters())
+            self.__operations.append((operation, operation_parameters))
+
+        # No operation with the specified name in this scope
+        else:
+            if operation_parameters.Has("module_name"):
+                module_name = operation_parameters["module_name"].GetString()
+                module = import_module(f"KratosMultiphysics.HDF5Application.core.{module_name}")
+                operation = module.Create(operation_parameters)
+
+            if operation is None:
+                raise ValueError(f"Invalid operation type '{operation_parameters['operation_type'].GetString()}'. Available options: {[self.__CamelToSnake(op.__name__) for op in GetSubclasses(IOOperation)]}")
+            else:
+                self.__operations.append((operation, operation_parameters))
+
+    @staticmethod
+    def GetDefaultParameters() -> KratosMultiphysics.Parameters:
+        return KratosMultiphysics.Parameters("""{
+            "model_part_name" : "",
+            "list_of_operations" : [],
+            "io_settings" : {}
+        }""")
+
+    @staticmethod
+    def __SnakeToCamel(string: str) -> str:
+        return "".join(part.title() for part in string.split('_'))
+
+    @staticmethod
+    def __CamelToSnake(string: str) -> str:
+        capital_indices = [0] + [index for index, char in enumerate(string) if char.isupper()] + [len(string)]
+        return "_".join(string[begin:end].lower() for begin, end in zip(capital_indices[:-1], capital_indices[1:]) if begin != end)
```

Comment:
I would use the one from the core and implement the one which is missing in the core
https://github.com/KratosMultiphysics/Kratos/blob/fd0c366d716422c4e1abbab18ae050faedb64df8/kratos/utilities/string_utilities.h#L58

Reply:
but ...
but ...
this one is so ugly and unreadable xD

Reply:
@sunethwarna I'll replace these when #10695 gets merged.